src/main/java/teammates/storage/api/FeedbackResponsesDb.java

```diff
@@ -102,7 +102,7 @@ public FeedbackResponse getFeedbackResponseEntityWithCheck (
         FeedbackResponse fr = 
                 getFeedbackResponseEntity(feedbackQuestionId, giverEmail, receiverEmail);
         
-        if (fr == null) {
+        if (fr == null || JDOHelper.isDeleted(fr)) {
```

Comment:
Isn't this redundant since <code>getFeedbackResponseEntity</code> already performs this check?